src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -21,51 +105,1972 @@ const ActorInit Boss_Fd_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/BossFd_Draw.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_fd.c", 0x11ED);
+    if (this->skinSegments != 0) {
+        gSPSegment(oGfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(D_808D1BB8[this->eyeState]));
+    }
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (s16)this->bodyTex1Scroll, (s16)this->bodyTex1Rot, 0x20,
+                                0x20, 1, (s16)this->bodyTex2Scroll, (s16)this->bodyTex2Rot, 0x20, 0x20));
+    gDPSetPrimColor(oGfxCtx->polyOpa.p++, 0, 0, 0xFF, 0xFF, 0xFF, 0xFF);
+    gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0xFF, 0xFF, 0xFF, (s8)this->bodyTex2Opacity);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/func_808D00A4.s")
+    osSyncPrintf("LH\n");
+    Matrix_Push();
+    segIndex = (D_808D1AC0[2] + this->leadBodySeg) % 100;
+    Matrix_Translate(this->bodySegsPos[segIndex].x, this->bodySegsPos[segIndex].y, this->bodySegsPos[segIndex].z, 0);
+    Matrix_RotateY(this->bodySegsRot[segIndex].y, 1);
+    Matrix_RotateX(-this->bodySegsRot[segIndex].x, 1);
+    Matrix_Translate(-13.0f, -5.0f, 13.0f, 1);
+    Matrix_Scale(this->actor.scale.x * 0.1f, this->actor.scale.y * 0.1f, this->actor.scale.z * 0.1f, 1);
+    SkelAnime_Draw(globalCtx, this->skelAnime2.skeleton, this->skelAnime2.limbDrawTbl, &func_808D00A4, 0, &this->actor);
+    Matrix_Pull();
+    osSyncPrintf("RH\n");
+    Matrix_Push();
+    segIndex = (D_808D1AC0[2] + this->leadBodySeg) % 100;
+    Matrix_Translate(this->bodySegsPos[segIndex].x, this->bodySegsPos[segIndex].y, this->bodySegsPos[segIndex].z, 0);
+    Matrix_RotateY(this->bodySegsRot[segIndex].y, 1);
+    Matrix_RotateX(-this->bodySegsRot[segIndex].x, 1);
+    Matrix_Translate(13.0f, -5.0f, 13.0f, 1);
+    Matrix_Scale(this->actor.scale.x * 0.1f, this->actor.scale.y * 0.1f, this->actor.scale.z * 0.1f, 1);
+    SkelAnime_Draw(globalCtx, this->skelAnime3.skeleton, this->skelAnime3.limbDrawTbl, &func_808D01C0, 0, &this->actor);
+    Matrix_Pull();
+    osSyncPrintf("BD\n");
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x0D, tempMat);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/func_808D01C0.s")
+    Matrix_Push();
+    for (i1 = 0; i1 < 0x12; i1++, tempMat++) {
+        segIndex = (D_808D1AC0[i1 + 1] + this->leadBodySeg) % 100;
+        Matrix_Translate(this->bodySegsPos[segIndex].x, this->bodySegsPos[segIndex].y, this->bodySegsPos[segIndex].z,
+                         0);
+        Matrix_RotateY(this->bodySegsRot[segIndex].y, 1);
+        Matrix_RotateX(-this->bodySegsRot[segIndex].x, 1);
+        Matrix_Translate(0.0f, 0.0f, 35.0f, 1);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
+        if (i1 < this->skinSegments) {
+            Matrix_Scale(1.0f + (Math_Sins((this->leadBodySeg * 5000.0f) + (i1 * 7000.0f)) *
+                                 *((f32*)this + 0xA0)), // this->bodyPulse
```

Comment:
`this->colChkInfo.displacement.y` ?

Reply:
No, it's this->bodyPulse. It's written like that because of some weird priority the compiler gives to the operands of floating point add. Temp variables fixed that problem, but then I couldn't get them to stack up correctly.

Reply:
my bad I forgot to multiply the offset by 4 because pointer arithmetics
`*(f32*)((s32)this + OFFSETOF(BossFd, bodyPulse))` looks just as bad but at least it shifts? does that work?
or `*((f32*)this + OFFSETOF(BossFd, bodyPulse)/sizeof(f32))` ?

Reply:
actually I found a much more plausible match. it seems the dereference is the key thing, and (*this).bodyPulse works.